plugins/core/hu.bme.mit.gamma.statechart.model/src/hu/bme/mit/gamma/statechart/derivedfeatures/StatechartModelDerivedFeatures.java

```diff
@@ -3544,4 +3557,67 @@ public static boolean hasNegatedContractStatechartAnnotation(Component component
 		return getComponentAnnotation(component, NegativeContractStatechartAnnotation.class) != null;
 	}
 	
+	protected static void _mapInterfaceInvariantToPort(Port port, BinaryExpression expression) {
```

Comment:
I believe all this is unnecessary. Wouldn't the following code snippet do what you want?
```
val interfaceParameterReferenceExpressions = expression.getSelfAndAllContentsOfType(InterfaceParameterReferenceExpression)
```